packages/wrangler/src/versions/secrets/index.ts

```diff
@@ -0,0 +1,332 @@
+	versionsSecretsDeleteOptions,
+} from "./delete";
+import { versionsSecretListHandler, versionsSecretsListOptions } from "./list";
+import { versionsSecretPutHandler, versionsSecretsPutOptions } from "./put";
+import type {
+	WorkerMetadata as CfWorkerMetadata,
+	WorkerMetadataBinding,
+} from "../../deployment-bundle/create-worker-upload-form";
+import type {
+	CfModule,
+	CfTailConsumer,
+	CfUserLimits,
+	CfWorkerInit,
+	CfWorkerSourceMap,
+} from "../../deployment-bundle/worker";
+import type { CommonYargsArgv } from "../../yargs-types";
+import type { File, SpecIterableIterator } from "undici";
+
+export function registerVersionsSecretsSubcommands(yargs: CommonYargsArgv) {
+	return yargs
+		.command(
+			"put <key>",
+			"Create or update a secret variable for a Worker",
+			versionsSecretsPutOptions,
+			versionsSecretPutHandler
+		)
+		.command(
+			"bulk [json]",
+			"Create or update a secret variable for a Worker",
+			versionsSecretsPutBulkOptions,
+			versionsSecretPutBulkHandler
+		)
+		.command(
+			"delete <key>",
+			"Delete a secret variable from a Worker",
+			versionsSecretsDeleteOptions,
+			versionsSecretDeleteHandler
+		)
+		.command(
+			"list",
+			"List the secrets currently deployed",
+			versionsSecretsListOptions,
+			versionsSecretListHandler
+		);
+}
+
+// Shared code
+export interface WorkerVersion {
+	id: string;
+	metadata: WorkerMetadata;
+	number: number;
+}
+
+export interface WorkerMetadata {
+	author_email: string;
+	author_id: string;
+	created_on: string;
+	modified_on: string;
+	source: string;
+}
+
+interface Annotations {
+	"workers/message"?: string;
+	"workers/tag"?: string;
+	"workers/triggered_by"?: string;
+}
+
+export interface VersionDetails {
+	id: string;
+	metadata: WorkerMetadata;
+	annotations?: Annotations;
+	number: number;
+	resources: {
+		bindings: WorkerMetadataBinding[];
+		script: {
+			etag: string;
+			handlers: string[];
+			placement_mode?: "smart";
+			last_deployed_from: string;
+		};
+		script_runtime: {
+			compatibility_date?: string;
+			compatibility_flags?: string[];
+			usage_model: "bundled" | "unbound" | "standard";
+			limits: CfUserLimits;
+		};
+	};
+}
+
+interface ScriptSettings {
+	logpush: boolean;
+	tail_consumers: CfTailConsumer[] | null;
+}
+
+interface CopyLatestWorkerVersionArgs {
+	accountId: string;
+	scriptName: string;
+	versionId: string;
+	secrets: { name: string; value: string; inherit?: boolean }[];
+	versionMessage?: string;
+	versionTag?: string;
+	sendMetrics?: boolean;
+	overrideAllSecrets?: boolean; // Used for delete - this will make sure we do not inherit any
+}
+
+// TODO: This is a naive implementation, replace later
+export async function copyWorkerVersionWithNewSecrets({
+	accountId,
+	scriptName,
+	versionId,
+	secrets,
+	versionMessage,
+	versionTag,
+	sendMetrics,
+	overrideAllSecrets,
+}: CopyLatestWorkerVersionArgs) {
+	// Grab the specific version info
+	const versionInfo = await fetchResult<VersionDetails>(
```

Comment:
do we need to handle the case when this API returns an error?

Reply:
It shouldn't ever generate an error, we already know the version exists. If it does, I'm happy treating that like all usual fatalerrors and having it reported to us